ASoC: SOF: ipc4-pcm: Pipe instances to dev_dbg in multi_pipeline_state()

jsarha opened the pull request:
Add a dev_dbg to sof_ipc4_set_multi_pipeline_state(). The debug print lists the pipeline instance numbers that are included in the SOF_IPC4_GLB_SET_PIPELINE_STATE message. Without this log its very hard to tell what pipelines are affected. This print is very helpful when analyzing SOF logs automatically.

Commit:
ASoC: SOF: ipc4-pcm: Pipe instances to dev_dbg in multi_pipeline_state()

Add a dev_dbg to sof_ipc4_set_multi_pipeline_state(). The debug print
lists the pipeline instance numbers that are included in the
SOF_IPC4_GLB_SET_PIPELINE_STATE message. Without this log its very
hard to tell what pipelines are affected. This print is very helpful
when analyzing SOF logs automatically.

Signed-off-by: Jyri Sarha <jyri.sarha@linux.intel.com>

ranj063 commented on sound/soc/sof/ipc4-pcm.c:
```diff
@@ -56,17 +56,40 @@ sof_ipc4_sps_to_time_info(struct snd_sof_pcm_stream *sps)
 						   state);
 
+	dev_dbg(sdev->dev, "pipeline instances%s",
+		sof_ipc4_set_multi_pipeline_state_debug(sdev, debug_buf, sizeof(debug_buf),
```
@jsarha can you please make it a little bit more verbose to add that these pipelines changed state from old_state to new_state?

jsarha replied:
@ranj063 , can you suggest something? My purpose was simply to add one log line telling the contents of the IPC message, right before the actual sending of the message. I did not pay too much attention in the logic how the the trigger list was built.

ranj063 replied:
something like "Pipelines 0, 1, 2 state switching to RUNNING\n"). You've got the pipeline IDs already and the function also has the new state thats being requested

jsarha replied:
Ok, I did almost exactly that, just left the ',' out to not waste space in stack or logs.

ranj063 replied:
Thank you, looks good now!

ujfalusi replied:
Why do we want different formats for prints if it is a single vs multiple pipeline that is changing state?
single pipeline (pipeline1 to state RUNNING):
`ipc4 set pipeline instance 1 state 3`
vs
multiple pipeline (pipeline1,2 to state RUNNING):
`Pipelines 1 2 state switching to  (3)`
or
`Pipelines 1 2 state switching to RUNNING (3)`

ujfalusi replied:
Why not for example:
single pipeline (pipeline1 to state RUNNING):
`Set pipeline 1 to state 3`
or Verbose:
`Set pipeline 1 to state 3 (RUNNING)`

multiple pipeline (pipeline1,2 to state RUNNING):
`Set pipelines 1, 2 to state 3`
or Verbose:
`Set pipelines 1, 2 to state 3 (RUNNING)`

jsarha replied:
To be honest I paid no attention to the other print since in single pipeline case, since I can read the instance id from the message header. In the multi-pipeline case the info is in the payload so a print is needed. 

But sure, I can harmonize the prints. But if its alright, I'll drop the ','s and '()'. They do not add any value but make things more complex.

Commit:
ASoC: SOF: ipc4: Add sof_ipc4_pipeline_state_str() for debugging

Add sof_ipc4_pipeline_state_str() to translate enum
sof_ipc4_pipeline_state into human readable form.

Signed-off-by: Jyri Sarha <jyri.sarha@linux.intel.com>

jsarha commented:
Seem I was too over confident. There was a compile error on previous version. Had to declare enum sof_ipc4_pipeline_state in ipc4-priv.h.

jsarha commented:
Just on space removed the sof_ipc4_set_multi_pipeline_state_debug() ads space in the beginnig, so we do not need it after "Pipelines".

jsarha commented:
... and the rebase.

ujfalusi commented on sound/soc/sof/ipc4-priv.h:
```diff
@@ -121,6 +121,10 @@ void sof_ipc4_update_cpc_from_manifest(struct snd_sof_dev *sdev,
 size_t sof_ipc4_find_debug_slot_offset_by_type(struct snd_sof_dev *sdev,
 					       u32 slot_type);
 
+
```
extra line added

Commit:
ASoC: SOF: ipc4-pcm: Harmonize sof_ipc4_set_pipeline_state() dbg print

Harmonize sof_ipc4_set_pipeline_state() dbg print with the new print
in sof_ipc4_set_multi_pipeline_state().

Signed-off-by: Jyri Sarha <jyri.sarha@linux.intel.com>

jsarha commented:
@ranj063 & @ujfalusi will this do now?

jsarha commented:
.. oops, changed =+ to +=

ranj063 commented on sound/soc/sof/ipc4-priv.h:
```diff
@@ -121,7 +121,6 @@ void sof_ipc4_update_cpc_from_manifest(struct snd_sof_dev *sdev,
 size_t sof_ipc4_find_debug_slot_offset_by_type(struct snd_sof_dev *sdev,
 					       u32 slot_type);
 
-
```
this is an unrelated change @jsarha

ranj063 replied:
if you remove this, this should be good to be merged

jsarha commented:
> @ranj063 & @ujfalusi will this do now?

What about now?

ujfalusi commented on sound/soc/sof/ipc4-pcm.c:
```diff
@@ -56,17 +56,41 @@ sof_ipc4_sps_to_time_info(struct snd_sof_pcm_stream *sps)
 		return sof_ipc4_set_pipeline_state(sdev, trigger_list->pipeline_instance_ids[0],
 						   state);
 
+	dev_dbg(sdev->dev, "Set pipelines %s to state %d %s",
```
"Set pipelines %s to state %d%s"
and sof_ipc4_pipeline_state_str(state) returns:
""
or
" (RUNNING)"

then the verbose print would look much better.

jsarha replied:
Sure, if you like the () so much 😉.

ujfalusi commented on sound/soc/sof/ipc4-pcm.c:
```diff
@@ -56,17 +56,41 @@ sof_ipc4_sps_to_time_info(struct snd_sof_pcm_stream *sps)
 		return sof_ipc4_set_pipeline_state(sdev, trigger_list->pipeline_instance_ids[0],
 						   state);
 
+	dev_dbg(sdev->dev, "Set pipelines %s to state %d %s",
```
You no longer need the space between %d and %s

jsarha replied:
Aargh... I squashed the change into wrong commit.

ujfalusi reviewed (approved):
@jsarha, thank you for the updates, looks good for me

jsarha commented:
@ranj063 could you click one more approve to get this merged?

lgirdwood reviewed (approved):
@ranj063 pls merge if you approve
